code/addons/docs/src/preset.ts

```diff
@@ -68,7 +68,10 @@ async function webpack(
   const mdxLoaderOptions = {
     // whether to skip storybook files, useful for docs only mdx or md files
     skipCsf: true,
-    remarkPlugins: [remarkSlug, remarkExternalLinks],
+    mdxCompileOptions: {
+      providerImportSource: '@storybook/addon-docs/mdx-react-shim',
```

Comment:
This value is not ALWAYS the same, because the user might install addon-docs directly or they might get it through the existence of addon-essentials.

So we need to:
- Add the same export to addon-essentials
- Make this configuration smarter, so it uses the right package (the one installed as a root dependency to the users' project.
- We add a preset-property called `mdxCompileOptions` options, which addon-docs AND addon-essentials BOTH set.
  When addon-docs is used stand-alone, that valuer should be used.
  When it's used through addon-essentials, the value from addon-essentials should be used.